contracts/ibc-reflect/src/contract.rs

```diff
@@ -172,10 +174,10 @@ pub fn ibc_channel_close(
     accounts(deps.storage).remove(channel_id.as_bytes());
 
     // transfer current balance if any (steal the money)
-    let amount = deps.querier.query_all_balances(&reflect_addr)?;
+    let amount = deps.querier.query_all_balances(reflect_addr.clone())?;
```

Comment:
After looking at the implementation, I wonder why you need a clone here now? Or is it just your choice?

I personally prefer using `&addr` where there is an `Into`, and letting it clone internally if needed. It is just a matter of taste, but it would be good to ensure that both forms work.

When I tried to change it, I got:

```
177 |     let amount = deps.querier.query_all_balances(&reflect_addr)?;
    |                               ^^^^^^^^^^^^^^^^^^ the trait `From<&Addr>` is not implemented for `std::string::String`
 ```

Reply:
Seems like one more From impl in std and I would be happy

Reply:
The problem I see here is that `.clone()` is relatively expensive and can often be avoided. Cloning `From` implementations hide that. But then again, the Rust standard library does this as well.

Reply:
Implemented in 82def5e1beb1dafbede8cc89d7bbd50d6dce7a94